plugins/neo4j-adapter/src/main/java/org/polypheny/db/adapter/neo4j/rules/graph/NeoLpgScan.java

```diff
@@ -20,37 +20,36 @@
 import org.polypheny.db.adapter.neo4j.NeoGraphImplementor;
 import org.polypheny.db.adapter.neo4j.rules.NeoGraphAlg;
 import org.polypheny.db.algebra.core.lpg.LpgScan;
-import org.polypheny.db.algebra.type.AlgDataTypeField;
-import org.polypheny.db.plan.AlgOptCluster;
+import org.polypheny.db.plan.AlgCluster;
 import org.polypheny.db.plan.AlgTraitSet;
-import org.polypheny.db.schema.TranslatableGraph;
-import org.polypheny.db.type.PolyType;
+import org.polypheny.db.schema.trait.ModelTrait;
 
-public class NeoLpgScan extends LpgScan implements NeoGraphAlg {
+public class NeoLpgScan extends LpgScan<NeoGraph> implements NeoGraphAlg {
 
 
     /**
      * Creates a {@link org.polypheny.db.adapter.neo4j.NeoConvention} of a {@link LpgScan}.
      *
      * @param cluster Cluster this expression belongs to
-     * @param traitSet Traits active for this node, including {@link org.polypheny.db.schema.ModelTrait#GRAPH}
+     * @param traitSet Traits active for this node, including {@link ModelTrait#GRAPH}
      */
-    public NeoLpgScan( AlgOptCluster cluster, AlgTraitSet traitSet, TranslatableGraph graph ) {
+    public NeoLpgScan( AlgCluster cluster, AlgTraitSet traitSet, NeoGraph graph ) {
         super( cluster, traitSet, graph );
     }
 
 
     @Override
     public void implement( NeoGraphImplementor implementor ) {
-        implementor.setGraph( (NeoGraph) getGraph() );
+        implementor.setGraph( entity );
 
-        if ( rowType.getFieldList().size() == 1 ) {
-            AlgDataTypeField field = rowType.getFieldList().get( 0 );
-            if ( field.getType().getPolyType() == PolyType.GRAPH ) {
-                implementor.setAll( true );
-            }
-        }
+        // if ( rowType.getFields().size() == 1 ) {
+        // AlgDataTypeField field = rowType.getFields().get( 0 );
+        // if ( field.getType().getPolyType() == PolyType.GRAPH ) {
+        implementor.setAll( true );
+        // }
+        //}
```

Comment:
Is this correct or a artifact?

Reply:
Yes, with the refactor this has become less annoying to manage. 
Removed the commented out parts.